src/share/poudriere/common.sh

```diff
@@ -2940,6 +2940,29 @@ jail_cleanup() {
 	export CLEANED_UP=1
 }
 
+download_from_repo() {
+	msg "Prefetching missing packages from pkg+http://pkg.freebsd.org/\${ABI}/${PACKAGE_BRANCH}"
+	cat >> ${MASTERMNT}/etc/pkg/poudriere.conf <<EOF
+FreeBSD: {
+        url: pkg+http://pkg.freebsd.org/\${ABI}/${PACKAGE_BRANCH};
+}
+EOF
+	umount ${UMOUNT_NONBUSY} ${MASTERMNT}/packages || \
+	    umount -f ${MASTERMNT}/packages
+	mount_packages
+	# only list packages which do not exists to prevent pkg from overwriting prebuilt packages
+	# XXX only work when PKG_EXT is the same as the upstream
+	(
+	while mapfile_read_loop "all_pkgs" pkgname originspec _ignored; do
```

Comment:
This assumes ` pwd`, please add this assertion at the top. Even if functionally it is guaranteed now I think the assertion is a good documentation of the subtle assumption.
```
[ "${PWD}" = "${MASTERMNT}/.p" ] || \                     
    err 1 "download_from_repo requires PWD=${MASTERMNT}/.p"
```